src/platform/FailSafeContext.cpp

```diff
@@ -85,19 +91,38 @@ void FailSafeContext::ScheduleFailSafeCleanup(FabricIndex fabricIndex, bool addN
 
 CHIP_ERROR FailSafeContext::ArmFailSafe(FabricIndex accessingFabricIndex, System::Clock::Timeout expiryLength)
 {
+    CHIP_ERROR err              = CHIP_NO_ERROR;
+    bool cancelTimersIfError    = false;
+    auto currentMonotonicTimeMs = System::SystemClock().GetMonotonicMilliseconds64();
+    if (!mFailSafeArmed)
+    {
+        System::Clock::Timeout maxCumulativeTimeout =
+            System::Clock::Milliseconds64(CHIP_DEVICE_CONFIG_MAX_CUMULATIVE_FAILSAFE_SEC * 1000);
+        SuccessOrExit(err = DeviceLayer::SystemLayer().StartTimer(maxCumulativeTimeout, HandleMaxCumulativeFailSafeTimer, this));
+        cancelTimersIfError = true;
+    }
+
     mFailSafeArmed = true;
     mFabricIndex   = accessingFabricIndex;
 
-    ReturnErrorOnFailure(DeviceLayer::SystemLayer().StartTimer(expiryLength, HandleArmFailSafeTimer, this));
-    ReturnErrorOnFailure(CommitToStorage());
-    ReturnErrorOnFailure(ConfigurationMgr().SetFailSafeArmed(true));
+    SuccessOrExit(err = DeviceLayer::SystemLayer().StartTimer(expiryLength, HandleArmFailSafeTimer, this));
+    SuccessOrExit(err = CommitToStorage());
+    SuccessOrExit(err = ConfigurationMgr().SetFailSafeArmed(true));
 
-    return CHIP_NO_ERROR;
+exit:
+
+    if (err != CHIP_NO_ERROR && cancelTimersIfError)
```

Comment:
While we are here, we should probably set `mFailSafeArmed` to false on error, right?  Otherwise we will be permanently stuck...  I know that's pre-existing.

Reply:
Made it so we only set `mFailSafeArmed` to true if we are successful, since I am not sure what we should do if we are re-arming and something fails